### PR TITLE
Split block_parser.dart and inline_parser.dart

### PR DESCRIPTION
**Pull request by chenzhiguang**
Now each syntax has a separate file, it makes the code much easier to maintain.

The following one property and two methods have been changed from private to public:
+ `InlineParser._encodeHtml` =>  `InlineParser.encodeHtml`
+ `InlineParser._pushDelimiter` => `InlineParser.pushDelimiter`
+ `LinkSyntax._createNode` => `LinkSyntax.createNode`

### CODE REVIEW

File: lib/src/block_syntaxes/block_html_syntax.dart [hunk 1]
```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2012, the Dart project authors.  Please see the AUTHORS file
```
[the remark this answers is not in the document]
Comment (kevmoo) [reply lib/src/block_syntaxes/block_html_syntax.dart:1]: yup!

[the remark this answers is not in the document]
Comment (chenzhiguang) [reply lib/src/block_syntaxes/block_html_syntax.dart:1]: So should I change to 2022 for all the new files?

**Comment by kevmoo** [comment]
Yes!

**Comment by chenzhiguang** [comment]
> Yes!
> […](#)

Fixed!

**Comment by chenzhiguang** [comment]
@srawlins Hi, will we merge it?

**Comment by srawlins** [comment]
I have not reviewed this yet. But hope to today.

**Comment by chenzhiguang** [comment]
Thanks a lot!

**Review by srawlins — APPROVED** [review]
Thanks!

**Pull request merged**